CHANGELOG_UNRELEASED.md

```diff
@@ -145,4 +147,6 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Infrastructure
 
+- `hierarchy.ml` now recognizes non-forgetful inheritance.
```

Comment:
I'm not sure I understand. It "recognizes" non-forgetful inheritance, but what does it do with it? Warn? Error? Tolerate given some special instructions?